### ISSUE TITLE
safeguard check that ensures that Markdown AST stays untouched

### ISSUE THREAD

**Issue by Borda** [executablebooks/mdformat#247]
### Describe the problem

Error: Could not format "/code/README.md".

The formatted Markdown renders to different HTML than the input Markdown. This
is likely a bug in mdformat. Please create an issue report here, including the
input Markdown: https://github.com/executablebooks/mdformat/issues

### Link to your repository or website

https://github.com/PyTorchLightning/pytorch-lightning/pull/8673

### Steps to reproduce

bot: https://results.pre-commit.ci/run/github/178626720/1627894852.sxZO2S7eSFu1ruIy9aSWfA
repo/PR: https://github.com/PyTorchLightning/pytorch-lightning/pull/8673

### The version of Python you're using

3.8

### Versions of your packages

```yaml
  - repo: https://github.com/executablebooks/mdformat
    rev: 0.7.7
    hooks:
      - id: mdformat
        additional_dependencies:
          - mdformat-gfm
          - mdformat-black
          - mdformat_frontmatter
        exclude: CHANGELOG.md
```

### Additional context

NOt really sure what is happening, so follow the instructions in the error message...

**Comment by hukkin** [comment]
Thanks for reporting! This is definitely a bug in the safeguard check that ensures that Markdown AST stays untouched.

If you want a temporary quick fix, changing this list in the README.md:
```markdown
  - _\** tests run on two NVIDIA P100_
  - _\*** tests run on Google GKE TPUv2/3_
  - _TPU py3.7 means we support Colab and Kaggle env._
```
to raw HTML should work:
```markdown
<ul>
<li><em>** tests run on two NVIDIA P100</em></li>
<li><em>*** tests run on Google GKE TPUv2/3</em></li>
<li><em>TPU py3.7 means we support Colab and Kaggle env.</em></li>
</ul>

```

I suspect the issue stems from the fact that the Markdown list is wrapped in some raw HTML tags.

I'll try to come up with a proper fix as soon as I can, but can't guarantee exactly when.

**Comment by Borda** [comment]
fixing as `**` -> `\*\*` and `***` -> `\*\*\*`

**Comment by hukkin** [comment]
That is the formatting that mdformat should produce yes, but I think the AST check still fails even if you fix that. Using the raw HTML I provided is probably the only way to make mdformat pass right now.

Btw there's an issue for adding a flag to skip validation https://github.com/executablebooks/mdformat/issues/50
That could also be useful as a temporary fix here, but is not implemented. Should be easy to implement though.

**Comment by hukkin** [comment]
I found a better workaround than the raw HTML: just remove leading spaces from the list and it formats fine.

Note to self: here's minimal Markdown to reproduce the issue

```markdown
  <center>

 - list item

  </center>
```

Seems like a fairly rare case: it doesn't reproduce using any HTML tag, requires leading whitespace for the HTML tags, and requires leading whitespace for the Markdown list.

**Comment by hukkin** [comment]
Thanks for reporting this! It should now be fixed in release 0.7.8 (uploaded as soon as CI is finished).
